src/docfx/lib/ProcessUtility.cs

```diff
@@ -326,23 +336,20 @@ public static Task WriteFile(string path, Action<Stream> write)
         public static async Task RunInsideMutex(string mutexName, Func<Task> action)
         {
             Debug.Assert(!string.IsNullOrEmpty(mutexName));
+            var lockPath = Path.Combine(AppData.MutexRoot, HashUtility.GetMd5Hash(mutexName));
 
-            // avoid the RunInsideMutex to be nested used
-            // doesn't support to require a lock before releasing a lock
-            // which may cause deadlock
-            if (t_innerCall.Value != null)
+            // avoid the RunInsideMutex to be nested used which cause deadlock
+            if (t_innerCall.Value != null && t_innerCall.Value == lockPath)
```

Comment:
can this be done in another PR? to allow fine grained nest detection? This should be a stack/list to detect deadlocks.

Reply:
ok